traits/constants.py

```diff
@@ -0,0 +1,195 @@
+#  Copyright (c) 2019, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in enthought/LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+
+from enum import IntEnum
+
+
+class TraitKind(IntEnum):
+    """ These determine the getters and setters used by the cTrait instance. """
+
+    #: A standard trait (validates and notifies).
+    trait = 0
+
+    #: A standard Python attribute (no validation or notification).
+    #: This sets valus into the object dict, so is not suitable for use with
+    #: standard Python descriptors - generally you want ``generic``.
+    python = 1
+
+    #: An event trait (can't read value, only assign value).
+    event = 2
+
+    #: A delegated trait.
+    delegate = 3
+
+    #: A trait property.
+    property = 4
+
+    #: A trait that can neither be assigned or read.
+    disallow = 5
+
+    #: A trait that can be assigned once, and after that only read (no
+    #: validation or notification).
+    read_only = 6
+
+    #: A trait whose value is constant (can only be read).
+    constant = 7
+
+    #: A standard Python attribute or descriptors (no validation or
+    #: notification).
+    generic = 8
+
+
+class ValidateTrait(IntEnum):
+    """ These are indices into the ctraits.c validate_handlers array. """
+
+    #: A type check.
+    type = 0
+
+    #: An instance check.
+    instance = 1
+
+    #: A self-type check.
+    self_type = 2
+
+    #: An integer range check (unused).
+    int_range = 3
+
+    #: A floating-point range check.
+    float_range = 4
+
+    #: An enumerated item check.
+    enum = 5
+
+    #: A mapped item check.
+    map = 6
+
+    #: A complex fast validator check with multiple fast validators.
+    complex = 7
+
+    #: A slow validator (only used in complex validators).
+    slow = 8
+
+    #: A tuple type check.
+    tuple = 9
+
+    #: A prefix map item check.
+    prefix_map = 10
+
+    #: A coercable type check.
+    coerce = 11
+
+    #: A castable type check.
+    cast = 12
+
+    #: A function check.
+    function = 13
+
+    #: A Python-based validator.
+    python = 14
+
+    #: An adaptable object check.
+    adapt = 19
+
+    #: An integer check.
+    int = 20
+
+    #: A floating-point check.
+    float = 21
+
+
+class ComparisonMode(IntEnum):
+    """ Comparison mode.
+
+    Indicates when trait change notifications should be generated based upon
+    the result of comparing the old and new values of a trait assignment:
+
+    * 0 (no_compare): The values are not compared and a trait change
```

Comment:
This seems redundant with the `#:`-style docstrings below, but I couldn't see a way to persuade Sphinx to render the individual docstrings. :-(

Reply:
Related: https://github.com/sphinx-doc/sphinx/issues/5679

Reply:
Nothing to do here. Resolving.